pandas/tests/reshape/merge/test_merge_asof.py

```diff
@@ -1287,3 +1287,19 @@ def test_timedelta_tolerance_nearest(self):
         )
 
         assert_frame_equal(result, expected)
+
+    def test_int_type_tolerance(self):
+        # GH #28870
+
+        left = pd.DataFrame({"a": [0, 100, 200], "left_val": [1, 2, 3]})
+        right = pd.DataFrame({"a": [50, 150, 250], "right_val": [1, 2, 3]})
+        left["a"] = left["a"].astype(np.int32)
```

Comment:
Could you `pytest.mark.parameterize` over more int dtypes. e.g. `[int, "int64", "int32", "int16", "int8", "uint64", "uint32", "uint16", "uint8"]`